charms/worker/k8s/src/containerd.py

```diff
@@ -136,8 +140,8 @@ def __init__(self, *args, **kwargs):
             kwargs: construction keyword arguments
         """
         super().__init__(*args, **kwargs)
-        url = pydantic.AnyHttpUrl(self.url)
-        if not self.host and (host := url.host):
+        url = urlparse(self.url)
+        if not self.host and (host := url.netloc):
```

Comment:
this is a nice UX improvement.  It simplifies the configuratoin because `netloc` includes the port number whereas `host` doesn't

rather than the user having to supply:

```yaml
-   url: http://my.registry.io:5000
    host: my.registry.io
```

one can simply have
```yaml
-   url: http://my.registry.io:5000
```